redis/tests/test_cluster.rs

```diff
@@ -747,3 +760,132 @@ fn test_cluster_split_multi_shard_command_and_combine_arrays_of_values() {
     let result = cmd.query::<Vec<String>>(&mut connection).unwrap();
     assert_eq!(result, vec!["foo-6382", "bar-6380", "baz-6380"]);
 }
+
+#[cfg(feature = "tls-rustls")]
+mod mtls_test {
+    use super::*;
+    use crate::support::mtls_test::create_cluster_client_from_cluster;
+    use redis::ConnectionInfo;
+
+    #[test]
+    fn test_cluster_basics_with_mtls() {
+        let cluster = TestClusterContext::new_with_mtls(3, 0);
+
+        let client = create_cluster_client_from_cluster(&cluster, true).unwrap();
+        let mut con = client.get_connection().unwrap();
+
+        redis::cmd("SET")
+            .arg("{x}key1")
+            .arg(b"foo")
+            .execute(&mut con);
+        redis::cmd("SET").arg(&["{x}key2", "bar"]).execute(&mut con);
+
+        assert_eq!(
+            redis::cmd("MGET")
+                .arg(&["{x}key1", "{x}key2"])
+                .query(&mut con),
+            Ok(("foo".to_string(), b"bar".to_vec()))
+        );
+    }
+
+    #[test]
+    fn test_cluster_should_not_connect_without_mtls() {
+        let cluster = TestClusterContext::new_with_mtls(3, 0);
+
+        let client = create_cluster_client_from_cluster(&cluster, false).unwrap();
+        let connection = client.get_connection();
+
+        match cluster.cluster.servers.get(0).unwrap().connection_info() {
+            ConnectionInfo {
+                addr: redis::ConnectionAddr::TcpTls { .. },
+                ..
+            } => {
+                if connection.is_ok() {
+                    panic!("Must NOT be able to connect without client credentials if server accepts TLS");
+                }
+            }
+            _ => {
+                if let Err(e) = connection {
+                    panic!("Must be able to connect without client credentials if server does NOT accept TLS: {e:?}");
+                }
+            }
+        }
+    }
+
+    #[test]
```

Comment:
I don't know that we need the remaining tests here. As you mentioned, they don't add much value. I think just a basic test and then one asserting that requests fail if the client doesn't enable mTLS is sufficient.